keras_cv/models/object_detection/object_detection_base_model.py

```diff
@@ -98,22 +98,20 @@ def evaluate(
         return super().evaluate(x=dataset, _use_cached_eval_dataset=False, **kwargs)
 
     def encode_data(self, x, y):
-        y_for_metrics = y
-
         y = bounding_box.convert_format(
             y,
             source=self.bounding_box_format,
             target=self.label_encoder.bounding_box_format,
             images=x,
         )
-        y_training_target = self.label_encoder(x, y)
-        y_training_target = bounding_box.convert_format(
-            y_training_target,
+        box_targets, class_targets = self.label_encoder(x, y)
+        box_targets = bounding_box.convert_format(
+            box_targets,
             source=self.label_encoder.bounding_box_format,
             target=self.bounding_box_format,
             images=x,
         )
-        return x, (y_for_metrics, y_training_target)
+        return x, {"gt_boxes": box_targets, "gt_classes": class_targets}
```

Comment:
This is actually a deviation from the current behavior, we won't be able to use the COCO metrics anymore. 

This is fine if the callback works

Reply:
Yeah the callback works pretty well, and I really like the result output from pycoco